File: LaborAid/backend/app/schemas/user_portal.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel


class UserRecentItem(BaseModel):
    id: int
    kind: Literal["case", "document", "evidence", "research", "contract"]
    title: str
    updated_at: datetime


class UserOverview(BaseModel):
    cases: int
    documents: int
    evidence: int
    research: int
    contracts: int = 0
    vault_files: int = 0
    vault_bytes: int = 0
    recent: list[UserRecentItem]

File: LaborAid/frontend/src/lib/api/user-portal.ts
import apiClient, { TIMEOUT } from './client';

export type RecentKind = 'case' | 'document' | 'evidence' | 'research' | 'contract';

export interface UserRecentItem {
  id: number;
  kind: RecentKind;
  title: string;
  updated_at: string;
}

export interface UserOverview {
  cases: number;
  documents: number;
  evidence: number;
  research: number;
  contracts: number;
  vault_files: number;
  vault_bytes: number;
  recent: UserRecentItem[];
}

export const userPortalApi = {
  getOverview: () =>
    apiClient.get<UserOverview>('/user/overview', { timeout: TIMEOUT.short }).then((r) => r.data),
};
